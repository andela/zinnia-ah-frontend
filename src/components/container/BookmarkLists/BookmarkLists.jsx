import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Item, Icon } from 'semantic-ui-react';
import moment from 'moment';

// config
import { DEFAULT_ARTICLE_IMAGE_URL } from '../../../config/config';

// components
import Button from '../../presentationals/Button/Button';
import Title from '../../presentationals/Title/Title';
import { getBookmarks } from '../../../store/modules/profile';

export class BookmarkLists extends Component {
  componentDidMount() {
    this.props.getBookmarks();
  }
  render() {
    const { bookmarks } = this.props;
    return (
      <div className="post-lists">
        {bookmarks.length === 0 ? (
          <p>You have no bookmarks</p>
        ) : (
          bookmarks.map(bookmark => (
            <Item.Group key={bookmark.id}>
              <Item>
                <Item.Image
                  src={bookmark.imageThumbnail || DEFAULT_ARTICLE_IMAGE_URL}
                />
                <Item.Content>
                  <div className="d-flex">
                    <div style={{ marginRight: 'auto' }}>
                      <Title
                        content={bookmark.title}
                        className="title-article-lg"
                      />
                    </div>
                    <div style={{ marginRight: '0.8rem' }}>
                      <p
                        style={{
                          fontSize: '1rem',
                          width: '100%',
                        }}
                      >
                        {bookmark.readTime} mins read
                      </p>
                    </div>
                  </div>
                  <Item.Description
                    style={{
                      marginBottom: 'auto',
                    }}
                  >
                    <p
                      style={{
                        fontSize: '1.125rem',
                        lineHeight: '1.375rem',
                        fontWeight: '200',
                      }}
                    >
                      {bookmark.description}
                    </p>
                  </Item.Description>
                  <Item.Extra>
                    <div className="d-flex justify-content-between">
                      <p style={{ fontSize: '1rem' }}>
                        {moment().from(bookmark.createdAt, 'YYYYMMDD')}
                      </p>
                      <div>
                        <div>
                          <Button
                            className="btn-transparent"
                            type="button"
                            value={
                              <Icon
                                name="bookmark"
                                style={{
                                  fontSize: '1.25rem',
                                  margin: '0',
                                  color: '#000000',
                                }}
                              />
                            }
                          />
                        </div>
                      </div>
                    </div>
                  </Item.Extra>
                </Item.Content>
              </Item>
            </Item.Group>
          ))
        )}
      </div>
    );
  }
}

BookmarkLists.propTypes = {
  bookmarks: PropTypes.array,
  getBookmarks: PropTypes.func,
};

export const mapStateToProps = state => ({
  error: state.profile.error,
  isLoading: state.profile.isLoading,
  bookmarks: state.profile.bookmarks,
});

export default connect(
  mapStateToProps,
  { getBookmarks },
)(BookmarkLists);
