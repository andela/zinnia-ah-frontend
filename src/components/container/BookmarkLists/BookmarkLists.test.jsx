import React from 'react';
import { mount } from 'enzyme';

import { BookmarkLists, mapStateToProps } from './BookmarkLists';

const props = {
  getBookmarks: jest.fn(),
  bookmarks: [
    {
      id: '2345rfghujhjk',
      userId: '5a6fab9c-4wf9-4bwef5-973c-5a371165cd57',
      slug: 'Hello-Article-1-db9cwefifhcoiwnf2e4-4d31-9015-288cf41d4092',
      title: 'EMMSDefqefrticle',
      description: 'Emmawkfvgwbvu wubvw djkvhis article',
      body: 'Another cgdsibcod cwdjcb djcubd cjdbc jdwbcikcuic dBody',
      imageThumbnail: 'https://dummy.jpg',
      readTime: '0',
      createdAt: '2019-05-16T11:32:52.024Z',
    },
  ],
  isLoading: false,
  error: {},
};
const propsWithoutImage = {
  getBookmarks: jest.fn(),
  bookmarks: [
    {
      id: '2345rfghujhjk',
      userId: '5a6fab9c-4wf9-4bwef5-973c-5a371165cd57',
      slug: 'Hello-Article-1-db9cwefifhcoiwnf2e4-4d31-9015-288cf41d4092',
      title: 'EMMSDefqefrticle',
      description: 'Emmawkfvgwbvu wubvw djkvhis article',
      body: 'Another cgdsibcod cwdjcb djcubd cjdbc jdwbcikcuic dBody',
      imageThumbnail: '',
      readTime: '0',
      createdAt: '2019-05-16T11:32:52.024Z',
    },
  ],
  isLoading: false,
  error: {},
};

const propsWithoutBookmarks = {
  getBookmarks: jest.fn(),
  bookmarks: [],
  isLoading: false,
  error: {},
};

describe('BookmarkLists', () => {
  it('should render correctly', () => {
    const component = mount(<BookmarkLists {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('mounts correctly without image', () => {
    const shallowWrapper = mount(<BookmarkLists {...propsWithoutImage} />);
    expect(shallowWrapper).toMatchSnapshot();
  });

  it('mounts correctly without bookamrks', () => {
    const shallowWrapper = mount(<BookmarkLists {...propsWithoutBookmarks} />);
    expect(shallowWrapper).toMatchSnapshot();
  });

  it('mounts correctly', () => {
    const DEFAULT_STATE = {
      profile: {
        error: {},
        isLoading: false,
        bookmarks: [],
      },
    };
    const action = mapStateToProps(DEFAULT_STATE);
    expect(action).toEqual(DEFAULT_STATE.profile);
  });
});
