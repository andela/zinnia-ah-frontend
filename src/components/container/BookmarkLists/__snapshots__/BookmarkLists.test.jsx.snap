// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookmarkLists mounts correctly without bookamrks 1`] = `
<BookmarkLists
  bookmarks={Array []}
  error={Object {}}
  getBookmarks={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  isLoading={false}
>
  <div
    className="post-lists"
  >
    <p>
      You have no bookmarks
    </p>
  </div>
</BookmarkLists>
`;

exports[`BookmarkLists mounts correctly without image 1`] = `
<BookmarkLists
  bookmarks={
    Array [
      Object {
        "body": "Another cgdsibcod cwdjcb djcubd cjdbc jdwbcikcuic dBody",
        "createdAt": "",
        "description": "Emmawkfvgwbvu wubvw djkvhis article",
        "id": "2345rfghujhjk",
        "imageThumbnail": "",
        "readTime": "0",
        "slug": "Hello-Article-1-db9cwefifhcoiwnf2e4-4d31-9015-288cf41d4092",
        "title": "EMMSDefqefrticle",
        "userId": "5a6fab9c-4wf9-4bwef5-973c-5a371165cd57",
      },
    ]
  }
  error={Object {}}
  getBookmarks={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  isLoading={false}
>
  <div
    className="post-lists"
  >
    <ItemGroup
      key="2345rfghujhjk"
    >
      <div
        className="ui items"
      >
        <Item>
          <div
            className="item"
          >
            <ItemImage
              src="https://react.semantic-ui.com/images/wireframe/image.png"
            >
              <Image
                as="img"
                src="https://react.semantic-ui.com/images/wireframe/image.png"
                ui={false}
                wrapped={true}
              >
                <div
                  className="image"
                >
                  <img
                    src="https://react.semantic-ui.com/images/wireframe/image.png"
                  />
                </div>
              </Image>
            </ItemImage>
            <ItemContent>
              <div
                className="content"
              >
                <div
                  className="d-flex"
                >
                  <div
                    style={
                      Object {
                        "marginRight": "auto",
                      }
                    }
                  >
                    <Title
                      className="title-article-lg"
                      content="EMMSDefqefrticle"
                    >
                      <h1
                        className="title title-article-lg"
                      >
                        EMMSDefqefrticle
                      </h1>
                    </Title>
                  </div>
                  <div
                    style={
                      Object {
                        "marginRight": "0.8rem",
                      }
                    }
                  >
                    <p
                      style={
                        Object {
                          "fontSize": "1rem",
                          "width": "100%",
                        }
                      }
                    >
                      0
                       mins read
                    </p>
                  </div>
                </div>
                <ItemDescription
                  style={
                    Object {
                      "marginBottom": "auto",
                    }
                  }
                >
                  <div
                    className="description"
                    style={
                      Object {
                        "marginBottom": "auto",
                      }
                    }
                  >
                    <p
                      style={
                        Object {
                          "fontSize": "1.125rem",
                          "fontWeight": "200",
                          "lineHeight": "1.375rem",
                        }
                      }
                    >
                      Emmawkfvgwbvu wubvw djkvhis article
                    </p>
                  </div>
                </ItemDescription>
                <ItemExtra>
                  <div
                    className="extra"
                  >
                    <div
                      className="d-flex justify-content-between"
                    >
                      <p
                        style={
                          Object {
                            "fontSize": "1rem",
                          }
                        }
                      >
                        Invalid date
                      </p>
                      <div>
                        <div>
                          <Button
                            className="btn-transparent"
                            type="button"
                            value={
                              <Icon
                                as="i"
                                name="bookmark"
                                style={
                                  Object {
                                    "color": "#000000",
                                    "fontSize": "1.25rem",
                                    "margin": "0",
                                  }
                                }
                              />
                            }
                          >
                            <button
                              className="btn btn-transparent"
                              type="button"
                            >
                              <Icon
                                as="i"
                                name="bookmark"
                                style={
                                  Object {
                                    "color": "#000000",
                                    "fontSize": "1.25rem",
                                    "margin": "0",
                                  }
                                }
                              >
                                <i
                                  aria-hidden="true"
                                  className="bookmark icon"
                                  onClick={[Function]}
                                  style={
                                    Object {
                                      "color": "#000000",
                                      "fontSize": "1.25rem",
                                      "margin": "0",
                                    }
                                  }
                                />
                              </Icon>
                            </button>
                          </Button>
                        </div>
                      </div>
                    </div>
                  </div>
                </ItemExtra>
              </div>
            </ItemContent>
          </div>
        </Item>
      </div>
    </ItemGroup>
  </div>
</BookmarkLists>
`;

exports[`BookmarkLists should render correctly 1`] = `
<BookmarkLists
  bookmarks={
    Array [
      Object {
        "body": "Another cgdsibcod cwdjcb djcubd cjdbc jdwbcikcuic dBody",
        "createdAt": "",
        "description": "Emmawkfvgwbvu wubvw djkvhis article",
        "id": "2345rfghujhjk",
        "imageThumbnail": "https://dummy.jpg",
        "readTime": "0",
        "slug": "Hello-Article-1-db9cwefifhcoiwnf2e4-4d31-9015-288cf41d4092",
        "title": "EMMSDefqefrticle",
        "userId": "5a6fab9c-4wf9-4bwef5-973c-5a371165cd57",
      },
    ]
  }
  error={Object {}}
  getBookmarks={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  isLoading={false}
>
  <div
    className="post-lists"
  >
    <ItemGroup
      key="2345rfghujhjk"
    >
      <div
        className="ui items"
      >
        <Item>
          <div
            className="item"
          >
            <ItemImage
              src="https://dummy.jpg"
            >
              <Image
                as="img"
                src="https://dummy.jpg"
                ui={false}
                wrapped={true}
              >
                <div
                  className="image"
                >
                  <img
                    src="https://dummy.jpg"
                  />
                </div>
              </Image>
            </ItemImage>
            <ItemContent>
              <div
                className="content"
              >
                <div
                  className="d-flex"
                >
                  <div
                    style={
                      Object {
                        "marginRight": "auto",
                      }
                    }
                  >
                    <Title
                      className="title-article-lg"
                      content="EMMSDefqefrticle"
                    >
                      <h1
                        className="title title-article-lg"
                      >
                        EMMSDefqefrticle
                      </h1>
                    </Title>
                  </div>
                  <div
                    style={
                      Object {
                        "marginRight": "0.8rem",
                      }
                    }
                  >
                    <p
                      style={
                        Object {
                          "fontSize": "1rem",
                          "width": "100%",
                        }
                      }
                    >
                      0
                       mins read
                    </p>
                  </div>
                </div>
                <ItemDescription
                  style={
                    Object {
                      "marginBottom": "auto",
                    }
                  }
                >
                  <div
                    className="description"
                    style={
                      Object {
                        "marginBottom": "auto",
                      }
                    }
                  >
                    <p
                      style={
                        Object {
                          "fontSize": "1.125rem",
                          "fontWeight": "200",
                          "lineHeight": "1.375rem",
                        }
                      }
                    >
                      Emmawkfvgwbvu wubvw djkvhis article
                    </p>
                  </div>
                </ItemDescription>
                <ItemExtra>
                  <div
                    className="extra"
                  >
                    <div
                      className="d-flex justify-content-between"
                    >
                      <p
                        style={
                          Object {
                            "fontSize": "1rem",
                          }
                        }
                      >
                        Invalid date
                      </p>
                      <div>
                        <div>
                          <Button
                            className="btn-transparent"
                            type="button"
                            value={
                              <Icon
                                as="i"
                                name="bookmark"
                                style={
                                  Object {
                                    "color": "#000000",
                                    "fontSize": "1.25rem",
                                    "margin": "0",
                                  }
                                }
                              />
                            }
                          >
                            <button
                              className="btn btn-transparent"
                              type="button"
                            >
                              <Icon
                                as="i"
                                name="bookmark"
                                style={
                                  Object {
                                    "color": "#000000",
                                    "fontSize": "1.25rem",
                                    "margin": "0",
                                  }
                                }
                              >
                                <i
                                  aria-hidden="true"
                                  className="bookmark icon"
                                  onClick={[Function]}
                                  style={
                                    Object {
                                      "color": "#000000",
                                      "fontSize": "1.25rem",
                                      "margin": "0",
                                    }
                                  }
                                />
                              </Icon>
                            </button>
                          </Button>
                        </div>
                      </div>
                    </div>
                  </div>
                </ItemExtra>
              </div>
            </ItemContent>
          </div>
        </Item>
      </div>
    </ItemGroup>
  </div>
</BookmarkLists>
`;
